### PR TITLE
Inject connections into constructor instead of using a method

### PR DESCRIPTION
**Pull request by odolbeau**
There is no reason to use a `addConnection` instead of pass them directly to the constructor.

### CODE REVIEW

**Comment by mfpierre** [comment]
It also exposes the clients through the container :+1:

File: Broker/PeclFactory.php [hunk 1]
```diff
@@ -14,12 +14,9 @@ class PeclFactory implements FactoryInterface
     protected $exchanges         = array();
     protected $amqpConnections   = array();
 
-    /**
-     * {@inheritDoc}
-     */
-    public function addConnection($name, array $connection)
+    public function __construct(array $connections = [])
```
Comment (mfpierre) [inline Broker/PeclFactory.php]: Phpdoc on the constructor ?

File: DependencyInjection/SwarrotExtension.php [hunk 2]
```diff
@@ -29,14 +29,10 @@ public function load(array $configs, ContainerBuilder $container)
...
-        $definition = $container->getDefinition($id);
-
         foreach ($config['connections'] as $name => $connectionConfig) {
-            $definition->addMethodCall('addConnection', array(
-                $name,
-                $connectionConfig
-            ));
+            $connections[$name] = $connectionConfig;
         }
+        $container->setParameter('swarrot.connections', $connections);
```
Comment (stof) [inline DependencyInjection/SwarrotExtension.php]: IMO, you should replace the argument instead of adding a parameter for that

Comment (odolbeau) [inline DependencyInjection/SwarrotExtension.php]: I closed this PR without merging it.
In fact, I needed this connections list to be able to reuse them in another bundle that's why I wished I had this parameter. :)

PR #67 solved my problem as it already expose a parameter containing the connection list.

Comment (stof) [inline DependencyInjection/SwarrotExtension.php]: well, removing the mutator from the interface still makes sense to me

**Pull request closed without merging**